src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,117 @@
-import React from 'react';
+import React, { useState, useCallback, FormEvent } from 'react';
+import cn from 'classnames';
+import { addComment } from '../api/comments';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { ErrorMessage } from '../types/errors';
+import { ErrorIcon } from './ErrorIcon';
+import { validateEmail } from '../types/validateEmail';
+
+interface Props {
+  post: Post | null;
+  setErrorMessage: React.Dispatch<React.SetStateAction<ErrorMessage | null>>;
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>;
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  post,
+  setErrorMessage,
+  setComments,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [comment, setComment] = useState('');
+  const [hasName, setHasName] = useState(true);
+  const [hasEmail, setHasEmail] = useState(true);
+  const [isEmailValid, setIsEmailValid] = useState(true);
+  const [hasComment, setHasComment] = useState(true);
+  const [isCommentAdding, setIsCommentAdding] = useState(false);
+
+  const addNewComment = useCallback(async () => {
+    setIsCommentAdding(true);
+
+    try {
+      const newComment: Comment = {
+        id: 0,
+        postId: post?.id ?? 0,
+        name,
+        email,
+        body: comment,
+      };
+
+      const fetchedComment = await addComment(newComment);
+
+      setComment('');
+      setComments(prevComments => [...prevComments, fetchedComment]);
+    } catch (error) {
+      setErrorMessage(ErrorMessage.ADD);
+    } finally {
+      setIsCommentAdding(false);
+      setHasName(true);
+      setHasEmail(true);
+      setHasComment(true);
+    }
+  }, [name, email, comment, post, setComments, setErrorMessage]);
+
+  const handleFormSubmit = (event: FormEvent) => {
+    event.preventDefault();
+    let isNotValid;
+
+    if (!name) {
+      setHasName(false);
+      isNotValid = true;
+    }
+
+    if (!email) {
+      setHasEmail(false);
+      isNotValid = true;
+    }
+
+    if (!validateEmail(email)) {
+      setIsEmailValid(false);
+      isNotValid = true;
```

Comment:
'isNotValid = true;' is duplicated, so you can add the one a condition and use in only once, for example:
if ( !email ||  !name || !validateEmail(email || !comment) {
 isNotValid = true;
}